engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -1868,6 +1847,65 @@ protected Void copyVolumeCallBack(AsyncCallbackDispatcher<VolumeServiceImpl, Cop
         return null;
     }
 
+    @Override
+    public boolean copyPoliciesBetweenVolumesAndDestroySourceVolumeAfterMigration(Event destinationEvent, Answer destinationEventAnswer, VolumeInfo sourceVolume,
+      VolumeInfo destinationVolume, boolean retryExpungeVolumeAsync) {
```

Comment:
@GutoVeronezi I think, it is better to separate copy policies and destroy volume.

Reply:
@sureshanaparti I extracted the code that destroy the source volume to a method and added tests to it; However I kept this method, calling both `copy policies` and `destroy volume` to facilitate and guide the process.